ui/components/multichain/pages/gator-permissions/components/review-gator-permission-item.tsx

```diff
@@ -214,24 +194,22 @@ export const ReviewGatorPermissionItem = ({
         expandedDetails: {
           initialAllowance: {
             translationKey: 'gatorPermissionsInitialAllowance',
-            value:
-              decimals === null
-                ? t('gatorPermissionUnknownTokenAmount')
-                : `${getDecimalizedHexValue(
-                    permission.data.initialAmount || '0x0',
-                    decimals,
-                  )} ${symbol}`,
+            value: loading
+              ? t('gatorPermissionUnknownTokenAmount')
+              : `${getDecimalizedHexValue(
+                  permission.data.initialAmount || '0x0',
+                  decimals,
+                )} ${symbol}`,
             testId: 'review-gator-permission-initial-allowance',
           },
           maxAllowance: {
             translationKey: 'gatorPermissionsMaxAllowance',
-            value:
-              decimals === null
-                ? t('gatorPermissionUnknownTokenAmount')
-                : `${getDecimalizedHexValue(
-                    permission.data.maxAmount || '0x0',
-                    decimals,
-                  )} ${symbol}`,
+            value: loading
+              ? t('gatorPermissionUnknownTokenAmount')
+              : `${getDecimalizedHexValue(
+                  permission.data.maxAmount || '0x0',
```

Comment:
Not strictly related to this change, but if maxAmount is undefined, shouldn't the value show "No maximum" or "Unlimited" rather than "0" as this is misleading.

Reply:
You’re correct. According to the logic in the code below, maxAmount should not be undefined. If that situation occurs, the expected value should be unlimited instead of 0.

https://github.com/MetaMask/snap-7715-permissions/blob/40c082766e86f82f7bb4498bacc39f2b6c96f36f/packages/gator-permissions-snap/src/permissions/erc20TokenStream/context.ts#L130